pynest/nest/lib/hl_api_types.py

```diff
@@ -547,6 +547,21 @@ def set(self, params, val=None):
     def tolist(self):
         return list(self)
 
+    def __getattr__(self, attr):
+        return self.get(attr)
+
+    def __setattr__(self, attr, value):
+        # check whether the attribute is a node property; need to manually
+        # exclude _datum since without it, get() does not work and we can not
+        # check whether attr is a node property
+        if attr != '_datum':
+            if attr in self.get().keys():
+                self.set({attr: value})
+                return  # if we have set the property in NEST, we are done
+
+        super().__setattr__(attr, value)  # if attr is not a node property,
+                                          # store it as regular attribute
```

Comment:
@jakobj This is super neat!

I don't think I quite understand why we need this last point tough. It is there so that we can set whatever we want on the GIDCollection if it is not a node property, do you have an example of what that could be? In principle I think this sounds a little unsafe, but it can be that I don't see all the things it can be used for.

Reply:
hmm, I thought it would still be nice to be able to set instance variables like so
```python
nodes = nest.Create('iaf_psc_alpha', 10)
nodes.my_variable = 'some value'
```
in order for this to work we need to use the `__setattr__` on the superclass as a far as I understand. however, you might be right that it could potentially lead to confusion as no error message would be raised if someone had a typo like `nodes.Vm = np.random.normal(-60., 1., 10)`.  this would just store create a new instance variable `Vm` but not actually affect the node properties in NEST.

Reply:
@jakobj: I quite like the idea of allowing the user to store additional properties for a `GIDCollection` purely on the Python level. However, I also see the danger you mention (that was the original reason for adding [access checks](https://github.com/nest/nest-simulator/blob/8c6843c50e8944e829b72a1c1555133c0f9164fe/sli/dict.h#L168) for SLI dictionaries). We should at least discuss this in a wider group and I recommend to remove it from this PR for now.